src/jni_core/src/lib.rs

```diff
@@ -823,6 +828,49 @@ pub extern "system" fn Java_com_risingwave_java_binding_Binding_rowClose<'a>(
     pointer.drop()
 }
 
+#[inline(never)]
+pub fn run_this_func_to_get_valid_ptr_from_java_binding() {}
+
+/// Send messages to the channel received by `CdcSplitReader`.
+/// If msg is null, just check whether the channel is closed.
+/// Return true if sending is successful, otherwise, return false so that caller can stop
+/// gracefully.
+#[no_mangle]
+pub extern "system" fn Java_com_risingwave_java_binding_Binding_sendCdcSourceMsgToChannel<'a>(
+    env: EnvParam<'a>,
+    channel: Pointer<'a, GetEventStreamJniSender>,
```

Comment:
Actually, I think the `channel` is more like a reference to the channel and we don't have its ownership, and we shouldn't drop the it here. 

Instead, we may have another jni method to close the channel, just like the close method of other structs such as iterator and row. On java side, we can have a class to wrap the channel pointer with a class, and implement `AutoClosable` for the class. At the beginning of the entry method `runJniDbzSourceThread` we can create such channel class from the pointer long, and wrap the rest code with a try-with-resource, so that the channel will be dropped no matter what exception is thrown. The current code has memory leak when any exception is thrown.

Reply:
I also think it is kind of weird to do the resource cleanup in this function, which should only handle message sending.